lib/shelley/src/Cardano/Wallet/Shelley/Pools.hs

```diff
@@ -98,13 +111,130 @@ import qualified Data.Map.Merge.Strict as Map
 import qualified Data.Map.Strict as Map
 import qualified Data.Text as T
 
--- | Stake Pool Data fields fetched from the node via LSQ
+--
+-- Stake Pool Layer
+--
+
+data StakePoolLayer = StakePoolLayer
+    { knownPools
+        :: IO [PoolId]
+
+    -- | List pools based given the the amount of stake the user intends to
+    -- delegate, which affects the size of the rewards and the ranking of the
+    -- pools.
+    , listStakePools
+        :: Coin
+        -> ExceptT ErrNetworkUnavailable IO [Api.ApiStakePool]
+    }
+
+newStakePoolLayer
+    :: GenesisParameters
+    -> NetworkLayer IO (IO Shelley) b
+    -> DBLayer IO
+    -> StakePoolLayer
+newStakePoolLayer gp nl db = StakePoolLayer
+    { knownPools = _knownPools
+    , listStakePools = _listPools
+    }
+  where
+    _knownPools
+        :: IO [PoolId]
+    _knownPools = do
+        tip <- getTip
+        let dummyCoin = Coin 0
+        res <- runExceptT $ map fst . Map.toList
+            . combineLsqData <$> stakeDistribution nl tip dummyCoin
+        case res of
+            Right x -> return x
+            Left _e -> return []
+
+    _listPools
+        :: Coin
+        -> ExceptT ErrNetworkUnavailable IO [Api.ApiStakePool]
+    _listPools userStake = do
+            tip <- liftIO getTip
+            lsqData <- combineLsqData <$> stakeDistribution nl tip userStake
+            chainData <- liftIO $ readDBPoolData db
+            return
+              . sortOn (Down . (view (#metrics . #nonMyopicMemberRewards)))
+              . map snd
+              . Map.toList
+              $ combineDBAndLsqData lsqData chainData
```

Comment:
_You can ignore this comment_

The mixed case style somewhat triggers some OCDs in me. 

```
combineDBAndLSQData
```

or 

```
combineDbAndLsqData
```

Mixing both looks weird.

Reply:
Went with `combineDbAndLsqData`. I can't find a guideline [here](https://github.com/input-output-hk/adrestia/wiki/Coding-Standards), but I thought `Lsq` was preferred to `LSQ`.

We do use `DB` a lot in our code.